### PR TITLE
CLI Execute Error prints to stderr

### PR DESCRIPTION
**Pull request by rigelrozanski**
instead of stdout as before, very useful for bash processing/error checking for CLI testing

### CODE REVIEW

File: cli/helper.go [hunk 1]
```diff
@@ -57,28 +57,39 @@ func RunWithArgs(cmd Executable, args []string, env map[string]string) error {
 // RunCaptureWithArgs executes the given command with the specified command line args
```
Comment (melekes) [inline cli/helper.go]: we should update the docs too

File: cli/helper.go [hunk 2]
```diff
@@ -57,28 +57,39 @@ func RunWithArgs(cmd Executable, args []string, env map[string]string) error {
...
 	// now run the command
 	err = RunWithArgs(cmd, args, env)
 
 	// and grab the stdout to return
-	w.Close()
-	output = <-outC
-	return output, err
+	wOut.Close()
+	wErr.Close()
+	stdout = <-outC
```
Comment (melekes) [inline cli/helper.go]: what if command only writes to stderr? will this code block forever?

Comment (rigelrozanski) [inline cli/helper.go]: I tried to make this happen with one pipe if that's what you're getting at not sure it's possible

Comment (rigelrozanski) [inline cli/helper.go]: Every example I've seen used recieves "" if standard output is empty so it doesn't lock

File: cli/helper.go (hunk 1 above)
Comment (rigelrozanski) [inline cli/helper.go]: done

File: cli/helper.go (hunk 2 above)
Comment (melekes) [inline cli/helper.go]: ok, good

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
+*.swp
```
Comment (melekes) [inline .gitignore]: https://robots.thoughtbot.com/global-gitignore

Comment (rigelrozanski) [inline .gitignore]: Thanks for the tip, just implemented on my machine

File: cli/helper.go [hunk 4]
```diff
@@ -54,31 +54,43 @@ func RunWithArgs(cmd Executable, args []string, env map[string]string) error {
...
 		var buf bytes.Buffer
-		// io.Copy will end when we call w.Close() below
-		io.Copy(&buf, r)
+		// io.Copy will end when we call wOut.Close() below
+		io.Copy(&buf, rOut)
 		outC <- buf.String()
 	}()
 
+	errC := make(chan string)
+	go func() {
```
Comment (ebuchman) [inline cli/helper.go]: Can we clean up the code dup here ?

File: cli/helper.go [hunk 5]
```diff
@@ -54,31 +54,43 @@ func RunWithArgs(cmd Executable, args []string, env map[string]string) error {
...
-// stdout along with any error returned from cmd.Execute()
-func RunCaptureWithArgs(cmd Executable, args []string, env map[string]string) (output string, err error) {
-	old := os.Stdout // keep backup of the real stdout
-	r, w, _ := os.Pipe()
-	os.Stdout = w
+// RunCaptureWithArgs executes the given command with the specified command
+// line args and environmental variables set. It returns string fields
+// representing output written to stdout and stderr, additionally any error
+// from cmd.Execute() is also returned
+func RunCaptureWithArgs(cmd Executable, args []string, env map[string]string) (stdout, stderr string, err error) {
```
Comment (ebuchman) [inline cli/helper.go]: Note this is breaking

**Comment by ebuchman** [comment]
Changelog please

File: cli/helper.go (hunk 5 above)
Comment (rigelrozanski) [inline cli/helper.go]: added to changelog

File: cli/helper.go (hunk 4 above)
Comment (rigelrozanski) [inline cli/helper.go]: Okay updated, let me know if sufficient, not sure we can compress this anymore

**Pull request merged**